docs/development/API/Dshot.md

```diff
@@ -0,0 +1,448 @@
+| 25-31 | Unused                                                                                            |
+| 32-63 | ESC signature                                                                                     |
+| 64    | CRC (same CRC as is used for telemetry)                                                           |
+
+### Calculating the CRC
+
+The checksum is calculated over the throttle value and the telemetry bit, so the “first” 12 bits our value in the following example:
+
+```
+crc = (value ^ (value >> 4) ^ (value >> 8)) & 0x0F;
+```
+
+Let’s assume we are sending a throttle value of 1046 - which is exactly half throttle and the telemetry bit is not set:
+
+```
+value = 100000101100
+(>>4) = 000010000010 # right shift value by 4
+(^) = 100010101110 # XOR with value
+(>>8) = 000000001000 # right shift value by 8
+(^) = 100010100110 # XOR with previous XOR
+(0x0F) = 000000001111 # Mask 0x0F
+(&) = 000000000110 # CRC
+```
+
+So the two bytes transmitted from flight-controller to ESC would be:
+
+```
+1000001011000110
+```
+
+We would put this signal on the wire to transmit the Dshot frame:
+
+![Dshot frame](./dshot_frame.webp)
+
+The green part are the throttle bits, blue is the telemetry bit and yellow the CRC checksum.
+
+### Arming sequence
+
+The arming sequenc might differ from implementation to implementation, but on most firmwares a 0 command is expected for a certain amount of time - 300ms on Bluejay for example - before the ESC goes into a state where it will accept actual throttle values.
+
+## Why is frame length important?
+
+The frame length is important because it indicates how fast the ESC can be updated. The shorter the frame length, the more often a frame can be sent per second. In other words the higher the bitrate, the more data we can send per second.
+
+This is then only limited by the loop speed of the flight-controller. Or the other way around, as we will see.
+
+> Let’s have a look at Dshot 300: A frame length of 106.72µs allows us to theoretically send 18768 full frames per second. Which results in a maximum frequency of around 18kHz.
+
+From this example **we can conclude that with a PID loop frequency of 8kHz we can’t exhaust Dshot 300**, so there is no real reason to run Dshot600 - at least if your PID loop frequency is 8kHz or less.
+
+But this is actually not the whole truth, since the flight controller spaces out the frames and locks it to the PID loop frequency. Dshot frame generation thus always runs at PID loop rate - this on the other hand means, that if you are running really high PID loop frequencies, you also need to run a high Dshot version.
+
+> Should you for example run a 32kHz loop, the flight controller will send DHSOT frames every 31.25µs - meaning you have to run at least Dshot600 in order to keep up.
+
+## What is ESC Telemetry?
+
+In the section about Frames I mentioned a telemetry bit. The flight controller uses this bit to request telemetry information from the ESC.
```

Comment:
```suggestion
In the section about Frames, it mentioned a telemetry bit. The flight controller uses this bit to request telemetry information from the ESC.
```

Reply:
Seems hard to say if this is better or worse.